src/Block/Adminhtml/Order/View/View.php

```diff
@@ -0,0 +1,150 @@
+use Shopgate\Import\Block\Adminhtml\Order\DataHydrator;
+
+class View extends Template
+{
+
+    /**
+     * @var OrderRepository
+     */
+    private $orderRepository;
+    /**
+     * @var array
+     */
+    private $whitelist;
+    /**
+     * @var Encoder
+     */
+    private $jsonDecoder;
+
+    /**
+     * @param OrderRepository $orderRepository
+     * @param Template\Context $context
+     * @param Encoder $jsonDecoder
+     * @param array $whitelist - our whitelisted data to print, see di.xml
+     * @param array $data
+     */
+    public function __construct(
+        OrderRepository $orderRepository,
+        Template\Context $context,
+        Encoder $jsonDecoder,
+        array $whitelist = [],
+        array $data = []
+    ) {
+        parent::__construct($context, $data);
+        $this->orderRepository = $orderRepository;
+        $this->whitelist = $whitelist;
+        $this->jsonDecoder = $jsonDecoder;
+    }
+
+    /**
+     * Checks if it's a Shopgate order
+     *
+     * @return bool
+     */
+    public function isShopgateOrder(): bool
+    {
+        return $this->getShopgateOrder() && !$this->getShopgateOrder()->isEmpty();
+    }
+
+    /**
+     * Returns clean payment info
+     *
+     * @return array
+     */
+    public function getPaymentInfos(): array
+    {
+        $payment = $this->jsonDecoder->decode($this->getShopgateOrder()->getReceivedData())['payment_infos'] ?? [];
+
+        return (new DataHydrator($payment))
+            ->filterWhitelist($this->whitelist['payment_infos'] ?? [])
+            ->removeEmpty()
+            ->readableKeys()
+            ->getData();
+    }
+
+    /**
+     * Prints all payment info
+     *
+     * @return string
+     */
+    public function printPaymentInfo(): string
+    {
+        $paymentInfo = $this->getPaymentInfos();
+        if (!$paymentInfo) {
+            return '';
+        }
+
+        return $this->printList($paymentInfo);
+    }
+
+    /**
+     * Prints lists under the main one
+     *
+     * @param array $list
+     * @param ?string $title - meant for sub-lists only, check <li> tags
+     * @return string
+     */
+    private function printList(array $list, ?string $title = null): string
+    {
+        $html = $title ? "<li><strong>$title</strong>" : ''; // if sublist
+        $html .= '<ul>';
+        foreach ($list as $key => $value) {
+            if (is_array($value)) {
+                $html .= $this->printList($value, $key);
+            } else {
+                $html .= "<li><strong>$key</strong>: $value</li>";
+            }
+        }
+        $html .= '</ul>';
+        $html .= $title ? '</li>' : '';
+
+        return $html;
+    }
+
+    /**
+     * Retrieves SG order from DB
+     *
+     * @return null|OrderInterface|Order
+     */
+    private function getShopgateOrder()
+    {
+        $id = $this->getRequest()->getParam('order_id');
+
+        try {
+            return $this->orderRepository->getByMageOrder($id);
```

Comment:
Or maybe cache this right here. Seems the function may be called several times during one request. Data likely won't change (or if at all, not dramatically) during one request's lifetime?

Reply:
yup, safe to assume, it can be cached